### DISCUSSION TITLE
Fixing redirect after page reload

### DISCUSSION THREAD

**Discussion by mrkvon** [0l5en/pod-chat-client#29] [Ideas]
Hi, i've read in [your README](https://github.com/0l5en/pod-chat-client/#inruptsolid-client-authn-browser)

> @inrupt/solid-client-authn-browser
>
> Doesn't work with the following versions: 1.13.1, 1.13.2, 1.13.3
> Wrong redirect URI seems to be used on browser reload.

Not sure if that's relevant, but have you seen [this doc about client-side routing redirect](https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/restore-session-browser-refresh/#use-session-restore-event-handler)? It solved similar-sounding issue for me. When you use client-side routing (`react-router-dom`), you need to put additional care to redirect. The link contains howto, and i can elaborate if needed.

**Comment by 0l5en** [comment]
Thanks for the hint. If you have a solution for the problem described, we would be very happy about it. As I already mentioned in the readme, a browser reload works up to version < 1.13.1. With all subsequent versions, a browser reload leads to an error message. This is very unfavorable for the user, as it is only possible to return to the start page if the cookies are deleted accordingly. You can find the relevant parts of the code here:

https://github.com/0l5en/pod-chat-client/blob/71907c57d6088deb0912cc68728e41af566f818e/src/views/ViewMain.tsx#L16-L25

The client will be initialized without any reditectUri set - like the documentation says it should use `window.location.href` per default.

**Reply by mrkvon** [reply]
Is this the issue you describe?
```
{"error":"invalid_request","error_description":"Mismatching redirect uri"}
```
This is not the error i assumed, so the link i provided won't be helpful, and i can't provide any further hints or solutions.

I did notice in inrupt library [changelog 1.13.1](https://github.com/inrupt/solid-client-authn-js/blob/main/CHANGELOG.md#bugfix-2):

> The [redirect URL] normalization step has now been removed, and the redirect provided by the user is sent to the OpenID Provider unchanged.

Somehow, the app sometimes sends redirect uri in form

`http://localhost:3001/` => success
and sometimes
`http://127.0.0.1:3001/?state=60b4545151c94a2a9aaaabcdef12345` => error

The latter happening to me after i restart the client development version. (`npm start`). But i was unable to reproduce or fix it consistently. So i'll leave it at that. Good luck!

**Comment by mrkvon** [comment]
Closing, as my initial suggestion was irrelevant to the issue. Feel free to reopen if you need...
